### PR TITLE
fix: duplicated upstream issue & job BC unclear logs

### PR DESCRIPTION
**Pull request by arinda-arif**
- fixing https://github.com/odpf/optimus/issues/715
- remove inaccurate logs saying upstream resolved externally while in fact there is no external optimus registered
- improve logs in external upstream resolver by adding namespace prefix
- improve logs when static upstream is specified but empty
- improve logs when unable to delete jobs while doing replace all

### CODE REVIEW

File: internal/store/postgres/job/job_repository.go [hunk 1]
```diff
@@ -359,7 +359,7 @@ func (JobRepository) toUpstreams(storeUpstreams []JobWithUpstream) ([]*job.Upstr
 	if err := me.ToErr(); err != nil {
 		return nil, err
 	}
-	return upstreams, nil
+	return job.Upstreams(upstreams).Deduplicate(), nil
```
Comment (sravankorumilli) [inline internal/store/postgres/job/job_repository.go:362]: check whether this deduplication is needed? & also the job inspect deduplication is not happening

**Review by sravankorumilli — CHANGES_REQUESTED** [review]
Added comments

File: internal/store/postgres/job/job_repository.go (hunk 1 above)
Comment (arinda-arif) [reply internal/store/postgres/job/job_repository.go]: have removed the multiple deduplication and add deduplication for job inspect case

**Comment by arinda-arif** [comment]
@sravankorumilli please check the recent changes:
- deduplicating from the root (for `bulk resolve` cases in the repository layer, for `single resolve` cases after merging static & inferred). the second deduplication is the same, after merging the unresolved with resolved ones.
- supports deduplicating unresolved upstreams

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**